### DISCUSSION TITLE
(Linux) Lazer + a realtime kernel?

### DISCUSSION THREAD

**Discussion by DaVince** [ppy/osu#30064] [Q&A] [answered]
Always in the pursuit of lower latency, a thought popped up in my head: trying out a realtime kernel. These are intended to make the maximum response time for a process predictable, which can greatly benefit latency on audio (and on input too, I've read?). So in theory, this _could_ provide faster response times in osu!.

I have also read that realtime kernels offer no real benefit to games in general. But osu! is a little special in that audio and input latency is that much more critical than the average game.

So what I'd like to know, before I even bother experimenting, is: has anyone already tried and benchmarked this? Or can any dev here tell me right out of the gate that it would/wouldn't make a difference?

**Comment by peppy** [comment, marked as answer]
It will probably not help. Everything in the chain would need to support this. And .NET definitely doesn't.

**Reply by DaVince** [reply]
Clear enough answer for me, thanks!
